lab_1/main.py

```diff
@@ -86,10 +183,43 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+    # Check for bad input
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profile_to_compare, dict)
+            or not isinstance(top_n, int)):
+        return None
+    # Find common words, while preserving the order
+    unknown_top = get_top_n_words(unknown_profile["freq"], top_n)
+    compare_top = get_top_n_words(profile_to_compare["freq"], top_n)
+    common = []
+    for token in compare_top:
+        if token in unknown_top:
```

Comment:
as far as I understand, you like list comprehensions
so can you write one instead of 194-197?